### PR TITLE
SCB-363 Remove legacy code

### PR DESCRIPTION
**Pull request by yangbor**
The support for legacy mode will be dropped, remove all the related
code from master branch.

Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SCB) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[SCB-XXX] Fixes bug in ApproximateQuantiles`, where you replace `SCB-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean install` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

### CODE REVIEW

File: docs/old_saga.md [hunk 1]
```diff
@@ -1,53 +0,0 @@
-# Previous Saga's Documentation
```
Comment (eric-lee-ltk) [inline docs/old_saga.md]: We should also remove the entry of this file in both `README.md` and `README_ZH.md`.

File: docs/old_saga.md [hunk 2]
```diff
@@ -1,53 +0,0 @@
...
-Enter the Saga root directory, run all service images using command,
-```
-docker-compose up
-```
-
-## Reference API
-See [Saga API](docs/api/api.md) for details.
-
-## Example
-See [Saga demo](https://github.com/apache/incubator-servicecomb-saga/tree/master/saga-demo) for details.
```
Comment (eric-lee-ltk) [inline docs/old_saga.md]: Maybe we should also rename the `saga-demo/pack-demo` to be something else like `saga-demo/booking` as it seems kinds of weird. Besides, after renaming the name, please update the demo entry in README and README_ZH.

File: docs/old_saga.md (hunk 1 above)
Comment (yangbor) [reply docs/old_saga.md]: Fixed

File: docs/old_saga.md (hunk 2 above)
Comment (yangbor) [reply docs/old_saga.md]: Fixed

**Pull request merged**